mti_nma/steps/singlecell/singlecell.py

```diff
@@ -34,62 +35,102 @@ def __init__(
-        """
+        '''
 
-        # Set reproducible random seed
+        import uuid
+        import numpy as np
         np.random.seed(666)
+        import pandas as pd
+        from tqdm import tqdm
+        from pathlib import Path
+        from aicsimageio import AICSImage, writers
 
-        # Set local directory to access raw data
-        raw_data_dir = Path('/Users/juliec/mti/raw_data')
+        if nsamples > 0:
 
-        # Load data from local directory, drawing random samples from among the cells
-        df = pd.read_csv(raw_data_dir / Path('ColonyMovie-ASCB2019.csv'), index_col=0)
-        df = df[['CellId', 'crop_raw', 'crop_seg']].sample(n=nsamples)
-        df = df.set_index('CellId')
+            self.manifest = pd.DataFrame([])
 
-        # Create directory to save data for this step in local staging
-        sc_data_dir = self.step_local_staging_dir / Path('singlecell_data')
-        sc_data_dir.mkdir(parents=True, exist_ok=True)
+            print("Loading data from LabKey...")
 
-        # Copy files to self.step_local_staging_dir and record in manifest
-        self.manifest = pd.DataFrame([])
-        for CellId in df.index:
-            shutil.copy(
-                src=raw_data_dir / Path('raw.ome.tif'),
-                dst=sc_data_dir / Path(f'{CellId}.raw.tif')
-            )
-            shutil.copy(
-                src=raw_data_dir / Path('segmentation.ome.tif'),
-                dst=sc_data_dir / Path(f'{CellId}.seg.tif')
+            df = query_data_from_labkey(CellLineId=cell_line_id) # 13 = Lamin
+
+            print(f"Number of FOVs available = {df.shape[0]}. Sampling {nsamples} FOVs now.")
+
+            df = df.sample(n=nsamples)
+
+            for fov_id in tqdm(df.index[:1]):
+
+                sx = df.PixelScaleX[fov_id]
+                sy = df.PixelScaleY[fov_id]
+                sz = df.PixelScaleZ[fov_id]
+
+                # C=-2 may not return the DNA channel for other cell lines. Need validation.
+                raw = AICSImage(df.ReadPathRaw[fov_id]).get_image_data('ZYX', S=0, T=0, C=-2)
+                seg = AICSImage(df.ReadPathSeg[fov_id]).get_image_data('ZYX', S=0, T=0, C= 0)
+
+                # Select one label from seg image at random
+                obj_label = np.random.randint(low=1, high=1+seg.max())
+
+                raw, seg = crop_object(
+                    raw = raw,
+                    seg = seg,
+                    obj_label = obj_label,
+                    isotropic = (sx,sy,sz))
+
+                # Create an unique id for this object
+                cell_id = uuid.uuid4().hex[:8]
+
+                # Save images
+                writer = writers.OmeTiffWriter(self.step_local_staging_dir.as_posix() + f'/{cell_id}.raw.tif')
+                writer.save(raw, dimension_order='ZYX')
+
+                writer = writers.OmeTiffWriter(self.step_local_staging_dir.as_posix() + f'/{cell_id}.seg.tif')
+                writer.save(seg, dimension_order='ZYX')
```

Comment:
Maybe throw a comment in the function docstring indicating that the channel order of the stored segs is ZYX